frontend/src/app/components/add-todo-list-modal/add-todo-list-modal.component.ts

```diff
@@ -50,22 +57,37 @@ export class AddTodoListModalComponent implements OnInit {
       this.timeHour = parseInt(time[0]);
       this.timeMinute = parseInt(time[1]);
       this.timePeriod = time[2].split(' ')[1];
+      this.todoItemTypes = data.todoItem.type;
     } else if (data.restoreTodoItem) {
       this.restoring = true;
       this.taskTitle = data.restoreTodoItem.title;
+      this.todoItemTypes = data.restoreTodoItem.type;
     }
   }
 
-  ngOnInit(): void {}
+  async ngOnInit(): Promise<void> {
+    this.userGroups = await this.groupService.getByUserAndProject(
+      this.userID,
+      this.projectID
+    );
+    if (this.data.group) {
+      this.selectedGroup = this.userGroups.find(
+        (g) => g.groupID === this.data.group.groupID
+      );
+    }
+  }
 
   async createTodoItem() {
+    console.log(this.selectedGroup);
     const todoItem: TodoItem = {
       todoItemID: generateUniqueID(),
       projectID: this.projectID,
       userID: this.userID,
       title: this.taskTitle,
+      groupID: this.selectedGroup ? this.selectedGroup.groupID : '',
```

Comment:
If there's no group selected, it would be preferred to leave this field null instead of an empty string. This would require the Todo model to make the `groupID` field optional.